content/articles/dockerized-prisma-postgres-api/index.md

```diff
@@ -0,0 +1,516 @@
+
+```bash
+docker-compose up
+```
+
+This command will run all the commands specified in the Docker file, pull the images for the set services, and wrap everything within a single container. Once the whole process is done, Docker will run the start command and run the API within the set container.
+
+![running-api](/engineering-education/dockerized-prisma-postgres-api/running-api.png)
+
+If you make some changes to the project files, run `docker-compose build` the re-run the `docker-compose up` command.
+
+### Testing the Docker API
+
+The app is running on Docker containers. Let's test if it works as expected. First, navigate to your Docker and check your running `dockerized-prisma-postgres-api` container.
+
+![prisma-postgres-api](prisma-postgres-api.png)
+
+Then hover over the `prisma-postgres-api` and open the integrated Docker API.
+
+![api-cli](/engineering-education/dockerized-prisma-postgres-api/api-cli.png)
+
+This will launch an interactive command line to run your API commands. In this case, you want to run the prima migrate command. This will allow your database to sync with the schema you have created. Go ahead and run `npx prisma migrate dev` inside his interactive Docker CLI.
+
+![docker-api](/engineering-education/dockerized-prisma-postgres-api/docker-cli.png)
+
+Let's now test the dockerized API. To do this, first, access the Postgres database using the pgadmin interface. This will let you see the data you add to your database server.
+
+Open `http://localhost:5555/` on your browser. The login using the username `pgadmin4@pgadmin.org` and password `pgadmin`, as set in them in the `dockercompose.yml` file.
+
+Finally, open Postman and start testing the API endpoints. Start by running the test GET route `http://localhost:3000`.
+
+![test-route](/engineering-education/dockerized-prisma-postgres-api/test-route.png)
+
+Test if the dockerized POST route is working using the route `http://localhost:3000/api/todos` as shown below:
+
+![post-api](/engineering-education/dockerized-prisma-postgres-api/post-api.png)
+
+Here is the sample input data:
+
+```json
+{
+"title": "Testing Dockerized Prisma API",
+"description": "a simple server that accesses Prisma to read its database schema a Docker container",
+"completed": false
+}
+```
+
+Once you hit the send button, a new todo will be created, and the below response will be recorded on Postman.
+
+![post-response](/engineering-education/dockerized-prisma-postgres-api/post-response.png)
+
+Finally, you can check this new todo using the GET `http://localhost:3000/api/todos`, as shown below:
+
+![post-api](/engineering-education/dockerized-prisma-postgres-api/post-api.png)
+
+### Conclusion
+
+This guide helped you run more about Prisma. In the end, you created a basic API using the PostgreSQL database. Then run it via Docker containers. Docker has enabled us to create the whole API without configuring PostgreSQL and Node.js on the local computer.
+
```

Comment:
Can you add a GitHub URL containing your source code?

Also, add a `Further reading` section that contains reference links